Distributions.cpp

```diff
@@ -0,0 +1,166 @@
+            exponential distibution of the form sigma_t * exp(-sigma_t * d).
+ @param     mat a Material object that contains information about the material
+ @param     group an int denoting the energy group of the neutron
+ @return    a randomly sampled distance in [0, infinity)
+*/
+double sampleDistance(Material* mat, int group) {
+    return -log(urand()) / mat->getSigmaT(group);
+}
+
+/*
+ @brief     function that samples the interaction type (0 = scattering,
+            1 = absorption)
+ @details   based on cross-sections, the interaction is sampled as scattering
+            (0) or absorption (1).
+ @param     mat a Material object that contains information
+            about the material
+ @param     group an int denoting the energy group of the neutron 
+ @return    an interaction type (0 = scattering, 1 = absorption)
+*/
+int sampleInteraction(Material* mat, int group) {
+    return (int) (urand() < mat->getSigmaA(group)
+                / mat->getSigmaT(group));
+}
+
+/*
+ @brief     function that samples a random location within a bounding box.
+ @details   a point is randomly and uniformally sampled in the bounding box 
+            provided in the input.
+ @param     bounds a Boundaries object containing the limits of the
+            bounding box
+ @return    _dist_location a vector that contains the coordinates of a point
+*/
+std::vector <double> sampleLocation(Boundaries bounds) {
+    std::vector <double> _dist_location(3);
+    for (int axis=0; axis<3; ++axis) {
+        double width = bounds.getSurfaceCoord(axis, 1)
+            - bounds.getSurfaceCoord(axis, 0);
+        double coord = bounds.getSurfaceCoord(axis, 0) + width * urand();
+        _dist_location[axis] = coord;
+    }
+    return _dist_location;
+}
+
+/*
+ @brief     function that samples the interaction type given an absorption
+            (0 = capture, 1 = fission)
+ @details   based on cross-sections, the interaction is sampled as capture
+            (0) or fission (1).
+ @param     mat a Material object that contains information about the material
+ @param     group contains the energy group of the neutron
+ @return    an interaction type (0 = capture, 1 = fission)
+*/
+int sampleFission(Material* mat, int group) {
+    int fission = urand() < mat->getSigmaF(group) / mat->getSigmaA(group);
+    return fission;
+}
+
+/*
+ @brief     samples the nunber of neutrons produced from a fission event
+ @param     mat a Material object that contains information about the material
+ @return    number of neutrons emitted from the sampled fission event
+*/
+int sampleNumFission(Material* mat) {
+    int lower = (int) mat->getNu();
+    int add = (int) (urand() < mat->getNu()-lower);
+    return lower + add;
+}
+
+/*
+ @brief     samples a neutron position in the fission bank
+ @param     fission_bank a Fission object containing neutron locations
+ @return    sampled neutron location
+*/
+std::vector <double> sampleFissionSite(
```

Comment:
@friedmud I agree. I made a ``Discributions`` file originally so @ljeure wouldn't have to worry about the sampling physics and so they would all be in one place, but moving forward it's a good idea to group things by their functionality in a true object-oriented fashion.